### PR TITLE
Allow to set segue flags for wasm-c-api JIT

### PR DESCRIPTION
**Pull request by wenyongh**

### CODE REVIEW

**Review by lum1n0us — APPROVED** [review]
LGTM

File: core/iwasm/common/wasm_c_api.c [hunk 1]
```diff
@@ -299,6 +299,11 @@ wasm_config_new(void)
 
     memset(config, 0, sizeof(wasm_config_t));
     config->mem_alloc_type = Alloc_With_System_Allocator;
+#if defined(os_writegsbase)
+    /* enable segue for all load/store operations */
+    config->segue_flags = 0x1F1F;
```
Comment (yamt) [inline core/iwasm/common/wasm_c_api.c:304]: i feel it's safer to disable segue stuff by default.
https://github.com/bytecodealliance/wasm-micro-runtime/issues/2736
how do you think?

File: core/iwasm/include/wasm_c_api.h [hunk 2]
```diff
@@ -200,6 +201,9 @@ wasm_config_set_mem_alloc_opt(wasm_config_t *, mem_alloc_type_t, MemAllocOption
 WASM_API_EXTERN own wasm_config_t*
 wasm_config_set_linux_perf_opt(wasm_config_t *, bool);
 
+WASM_API_EXTERN wasm_config_t*
+wasm_config_set_segue_flags(wasm_config_t *config, uint32_t segue_flags);
```
Comment (yamt) [inline core/iwasm/include/wasm_c_api.h:205]: do we have documentation about these flag bits?

File: core/iwasm/common/wasm_c_api.c (hunk 1 above)
Comment (wenyongh) [reply core/iwasm/common/wasm_c_api.c:304]: OK, maybe some targets don't support modifying GS register during execution, let's disable it by default.

File: core/iwasm/include/wasm_c_api.h (hunk 2 above)
Comment (wenyongh) [reply core/iwasm/include/wasm_c_api.h:205]: Not yet, how about adding comment here? For iwasm and wamrc, we had added comment in the command help info.

Comment (yamt) [reply core/iwasm/include/wasm_c_api.h:205]: > Not yet, how about adding comment here? For iwasm and wamrc, we had added comment in the command help info.

it sounds reasonable to me.

File: core/iwasm/common/wasm_c_api.c (hunk 1 above)
Comment (wenyongh) [reply core/iwasm/common/wasm_c_api.c]: Done.

File: core/iwasm/include/wasm_c_api.h (hunk 2 above)
Comment (wenyongh) [reply core/iwasm/include/wasm_c_api.h:213]: Done.

**Pull request merged**